Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -273,6 +275,7 @@ - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray
   // Fire all queued callbacks
   for (RCTLocationRequest *request in _pendingRequests) {
     request.successBlock(@[_lastLocationEvent]);
+    [request.timeoutTimer invalidate];
   }
   [_pendingRequests removeAllObjects];
```

Comment:
The new invalidate calls shouldn't be necessary because this should be invalidating the timers because all the requests should be dealloc'd at this point.  If that's not happening, there is probably a retain cycle or something?  cc @nicklockwood